### PR TITLE
Define a pull request template for GitHub

### PR DESCRIPTION
**Pull request by alex-mckenna**
It is relatively common in pull requests for small auxiliary changes
like updating copyright notices to be forgotten about. Using a PR
template would make this harder to do, and allow some pointers to
be provided to new contributers about what makes a useful message.

The template is somewhat minimal, I just based it on the things I like
about the messages in @martijnbastiaan's larger PRs. I don't think it
should be so long it feels like a chore to submit.

### CODE REVIEW

**Review by DigitalBrains1 — COMMENTED** [review]
It seems that while the body text is flowed properly even when the source contains line breaks, the fat lines behind `#` need to be one long line for it to format properly, i.e., a horizontal rule under the whole of the fat text rather than horizontal rules under every line.

**Comment by martijnbastiaan** [comment]
I agree with @DigitalBrains1 on the newlines. The rendering is a bit weird: https://github.com/clash-lang/clash-compiler/blob/fb29705a9de1b9ac8d056eb59005f3f7b8884937/.github/PULL_REQUEST_TEMPLATE.md. Did you intend to make [a comment](https://stackoverflow.com/questions/4823468/comments-in-markdown)?

I'd like to add some other points:

- CI will only run for contributors after maintainer approval
- Users can locally test with:
  -  `cabal v2-run clash-prelude:unittests`
  -  `cabal v2-run clash-prelude:doctests`
  - `cabal v2-run clash-lib`
  - `cabal v2-run testsuite -- -p  .Verilog` after `apt install iverilog`
  - (Purposely split up so `cabal` doesn't swallow output  and to make rerunning less time-consuming.)
  - (VHDL / SystemVerilog test suites are annoying to install dependencies for, so best to let the CI handle it.)

> I don't think it should be so long it feels like a chore to submit.

Agreed. We could add a `CONTRIBUTING.md` to elaborate. Perhaps instructions on running the tests should be covered there.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (DigitalBrains1) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
[comment]: # (Thank you for contributing to Clash. Please fill out this
	      template to describe your contribution and any outstanding
	      tasks / questions.)
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (DigitalBrains1) [inline .github/PULL_REQUEST_TEMPLATE.md:8]:
```suggestion
[comment]: # (Add a line of the form "Fixes: #xxxx" for each related issue
              closed by this pull request, where #xxxx is an issue number.)
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (DigitalBrains1) [inline .github/PULL_REQUEST_TEMPLATE.md:15]:
```suggestion
[comment]: # (Depending on the change, some of these points may not be
	      necessary. If so, leave the boxes unchecked so it is easier for a
	      reviewer to see what has been omitted.)
```

**Review by DigitalBrains1 — COMMENTED** [review]
Now that it no longer needs to be one line, how about limiting the length again? :-) At least, I presume this would be valid Markdown.

The GitHub web review editor is formatting my suggestions wrongly for some odd reason. In my suggestion, the first letters of the text are aligned vertically; everything is on the right of the opening parenthesis. But that's not how the GitHub web interface shows it to me. Very odd.

**Comment by martijnbastiaan** [comment]
IMO just leave it as is. People can use the corner thingy in their browser to resize to their preferred size.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (DigitalBrains1) [inline .github/PULL_REQUEST_TEMPLATE.md:6]:
```suggestion
Write a description of your PR here. Briefly describe the solution, taking more care with larger changes to outline the intention of the PR, or any compromises made in the final design. If there is no associated issue, it is helpful to also include the motivation for this PR.
```

**Review by DigitalBrains1 — COMMENTED** [review]
If we choose not to do line breaks in comments, let's then also not do line breaks in body text!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (martijnbastiaan) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: Agreed. Markdown = linebreaks on paragraphs only IMO.

**Review by martijnbastiaan — APPROVED** [review]

**Pull request merged**
